Adding manual scale a deployment (#54233)
kubernetes/website: content/en/docs/tasks/run-application/scale-deployment.md

---
reviewers:
title: Scale a Deployment Manually
content_type: Task
weight:
---

<!-- overview -->

This page shows how to manually scale a Kubernetes Deployment, and explains when manual scaling is appropriate compared to automatic scaling.

## {{% heading "prerequisites" %}}

{{< include "task-tutorial-prereqs.md" >}} {{< version-check >}}

<!-- lesson content -->

## Create a sample Deployment

If you do not already have a Deployment, you can create one for testing:

```shell
kubectl create deployment nginx --image=nginx
```

> To see whether the Deployment is created.

Reviewer: I would write:

To verify that the Deployment was created, run

```shell
kubectl get deployments
```

Sample of Deployment should be seen like this,

{{% code_sample file="application/deployment-update.yaml" %}}

> ## Procedure for manual scaling

Reviewer: I think ## Scaling a Deployment manually title is more descriptive

> Manual scaling adjusts the number of running Pods for a workload. By editing the Deployment manifest ==spec.replicas==.

Reviewer: Should be `spec.replicas` instead of ==spec.replicas==

> ### Scale up

Reviewer: ```suggestion
### Scale up a Deployment
```

Scaling up increases the capacity of an application to handle additional load.

```yaml
spec:
  replicas: 2
```

To increase the number of replicas:

```shell
kubectl scale deployment DEPLOYMENT_NAME --replicas=5
```

To verify the change in the Deployment:

```shell

> kubectl get deployment DEPLOYMENT_NAME

Reviewer: use angle bracket for placement
https://kubernetes.io/docs/contribute/style/style-guide/#use-angle-brackets-for-placeholders

kubectl get pods
```

When the replicas is increased from 2 to 5, Kubernetes creates three additional Pods to match the desired state and schedules them to available nodes while keeping the existing Pods unchanged.

### Scale down

Scaling down decreases the capcity of an application by decreasing the number of running Pods, allowing Kubernetes to terminate excess capacity while maintaining the desired state.

```yaml
spec:
  replicas: 5
```

To reduce the number of replicas:

```shell
kubectl scale deployment DEPLOYMENT_NAME --replicas=2
```

> When scaling down,Kubernetes terminates Pods according to the Pod termination lifecycle.

Reviewer: nit:  

```
When scaling down, Kubernetes terminates Pods according to the Pod termination lifecycle.
```

### Scale to zero

> Scale to zero set the replica count of a workload to 0.

Reviewer: I would say `Scaling a Deployment to zero sets the replica count to 0, which stops all Pods while keeping the Deployment object.`

```yaml
spec:
  replicas: 2
```

To stop all Pods managed by a Deployment:

```shell
kubectl scale deployment DEPLOYMENT_NAME --replicas=0
```

Scaling to zero keeps the Deployment object but removes all running Pods.

## Manual scaling versus Automatic scaling

| Manual scaling | Automatic scaling |
|----------------|-------------------|
| Developer updates replica count | Kubernetes adjusts replicas based on metrics |
| Predictable or temporary workload changes | Variable or unpredictable workloads |
| Use case load testing, planned traffic increases | Use case production services with fluctuating traffic |
| Explicit control over replicas | Autoscaling logic controls replicas |

> Kubernetes supports automatic scaling using a [HorizontalPodAutoscaler](/docs/tasks/run-application/horizontal-pod-autoscale/) (HPA). If a HPA is enabled for a Deployment, it may override manually set replica counts.

Reviewer: i would add a whatsnext headings 

## {{% heading "whatsnext" %}}

* Learn about [HorizontalPodAutoscaler](/docs/tasks/run-application/horizontal-pod-autoscale/)
* Read about [Pod disruption budgets](/docs/concepts/workloads/pods/disruptions/)

---
reviewers:
title: Scale a Deployment Manually
content_type: Task
weight: 90
---

<!-- overview -->

This page shows how to manually scale a Kubernetes Deployment, and explains when manual scaling is appropriate compared to automatic scaling.

## {{% heading "prerequisites" %}}

{{< include "task-tutorial-prereqs.md" >}} {{< version-check >}}

<!-- lesson content -->

## Create a sample Deployment

If you do not already have a Deployment, you can create one for testing:

```shell
kubectl create deployment nginx --image=nginx
```

To verify that the Deployment was created,run.

```shell
kubectl get deployments
```

The output is similar to this:

   ```
NAME               READY   UP-TO-DATE   AVAILABLE   AGE
nginx              1/1     1            1           10s
```

Sample of Deployment should be seen like this,

{{% code_sample file="application/deployment.yaml" %}}

> 

Reviewer: This part is a bit off if you ask me.

<img width="1390" height="942" alt="image" src="https://github.com/user-attachments/assets/fae14975-518f-4981-8916-cb02d268d0db" />

You can link to the deployment guide like i mentioned before,

## Creating a Deployment

If you do not already have a Deployment, follow the instructions in 
[Run a Stateless Application Using a Deployment](/docs/tasks/run-application/run-stateless-application-deployment/) 
to create one.

Alternatively, you can create a simple Deployment for testing:
```shell
kubectl create deployment nginx --image=nginx
```

To verify that the Deployment was created, run:
```shell
kubectl get deployments
```

The output is similar to:
```
NAME    READY   UP-TO-DATE   AVAILABLE   AGE
nginx   1/1     1            1           10s
```

## Scaling a Deployment manually

Manual scaling adjusts the number of running Pods for a workload. By editing the `spec.replicas` field in the Deployment manifest.

### Scale up a Deployment

Scaling up increases the capacity of an application to handle additional load.

```yaml
spec:
  replicas: 2

> ```

Reviewer: there is a disconnect here and the command instruction

To increase the number of replicas:

```shell
kubectl scale deployment <DEPLOYMENT_NAME> --replicas=4
```

{{< note >}}
Replace <DEPLOYMENT_NAME> with the name of your Deployment (for example, nginx).
{{</ note >}}

To verify the change in the Deployment:

```shell
kubectl get deployment <DEPLOYMENT_NAME>
kubectl get pods
```

The output similar to this:

{{% code_sample file="application/deployment-scale.yaml" %}}

When the replicas is increased from 2 to 5, Kubernetes creates three additional Pods to match the desired state and schedules them to available nodes while keeping the existing Pods unchanged.

### Scale down a Deployment

Scaling down decreases the capacity of an application by decreasing the number of running Pods, allowing Kubernetes to terminate excess capacity while maintaining the desired state.

```yaml
spec:
  replicas: 4
```

To reduce the number of replicas:

```shell
kubectl scale deployment <DEPLOYMENT_NAME> --replicas=2
```

The ouput similar to this:

> {{% code_sample file="application/deployment.yaml" %}}

Reviewer: this is not the expected outcome

## {{% heading "prerequisites" %}}

{{< include "task-tutorial-prereqs.md" >}} {{< version-check >}}

<!-- lesson content -->

## Create a sample Deployment

If you do not already have a Deployment, follow the instructions in [Run a Stateless Application Using a Deployment](/content/en/docs/tasks/run-application/run-stateless-application-deployment/) to create one.

Alternatively, you can create a simple Deployment for testing.

```shell
kubectl create deployment nginx --image=nginx
```

To verify that the Deployment was created, run:

```shell
kubectl get deployments
```

The output is similar to this:

```
NAME               READY   UP-TO-DATE   AVAILABLE   AGE
nginx              1/1     1            1           10s
```

## Scaling a Deployment manually

Manual scaling adjusts the number of running Pods for a workload by using commands such as `kubectl scale`.

### Scale up a Deployment

Scaling up increases the capacity of an application to handle additional load.

To increase the number of replicas:

```shell
kubectl scale deployment <deployment_name> --replicas=4
```

{{< note >}}
Replace <deployment-name> with the name of your Deployment (for example, nginx).

> {{</ note >}}

Reviewer 2: Inconsistent placeholder. The command uses underscores (<deployment_name>) but the note uses hyphens (<deployment-name>). The Kubernetes docs convention is <deployment-name> (hyphens).

Reviewer 3: Nit. Please address.

To verify the change in the Deployment:

```shell
kubectl get deployment <deployment_name>
kubectl get pods
```

> The output similar to this:

Reviewer 2: ```suggestion
The output is similar to this:
```

```
NAME    READY   UP-TO-DATE   AVAILABLE   AGE
nginx   4/4     4            4           3m16s
```
```
NAME                     READY   STATUS    RESTARTS   AGE
nginx-56c45fd5ff-xxxxx   1/1     Running   0          60s
nginx-56c45fd5ff-xxxxx   1/1     Running   0          60s
nginx-56c45fd5ff-xxxxx   1/1     Running   0          60s
nginx-56c45fd5ff-xxxxx   1/1     Running   0          60s
```

> When the replicas is increased from 2 to 4, Kubernetes creates three additional Pods to match the desired state and schedules them to available nodes while keeping the existing Pods unchanged.

Reviewer: should be two additional Pods

```suggestion
When the replicas is increased from 2 to 4, Kubernetes creates two additional Pods to match the desired state and schedules them to available nodes while keeping the existing Pods unchanged.
```

### Scale down a Deployment

Scaling down decreases the capacity of an application by decreasing the number of running Pods, allowing Kubernetes to terminate excess capacity while maintaining the desired state.

To reduce the number of replicas:

```shell
kubectl scale deployment <deployment_name> --replicas=2
```

{{< note >}}
Replace <deployment-name> with the name of your Deployment (for example, nginx).
{{</ note >}}

To verify the change in the Deployment:

```shell
kubectl get deployment <deployment_name>

> kubectl got pods

Reviewer 2: ```suggestion
kubectl get pods
```

```

The output is similar to this:

```
NAME    READY   UP-TO-DATE   AVAILABLE   AGE
nginx   2/2     2            2           7m16s
```
```
NAME                     READY   STATUS    RESTARTS   AGE
nginx-56c45fd5ff-xxxxx   1/1     Running   0          4m58s
nginx-56c45fd5ff-xxxxx   1/1     Running   0          7m27s
```

When scaling down, Kubernetes terminates Pods following the Pod termination lifecycle.

### Scale a Deployment to zero

Scale to zero set the replica count of a workload to 0.

To stop all Pods managed by a Deployment:

```shell
kubectl scale deployment <deployment_name> --replicas=0
```

{{< note >}}
Replace <deployment-name> with the name of your Deployment (for example, nginx).
{{</ note >}}

To verify the change in the Deployment:

```shell
kubectl get deployment <deployment_name>
kubectl get pods
```

The output is similar to this:

```
NAME    READY   UP-TO-DATE   AVAILABLE   AGE
nginx   0/0     0            0           19m
```
```
No resources found in default namespace.
```

> Scaling to zero keeps the Deployment object but removes all running Pods.

Reviewer: we could have way to verify this and show it result

kubectl get deployment <deployment_name>

## Manual scaling versus automatic scaling

| Manual scaling | Automatic scaling |
|----------------|-------------------|
| Developer updates replica count | Kubernetes adjusts replicas based on metrics |
| Predictable or temporary workload changes | Variable or unpredictable workloads |
| Use case load testing, planned traffic increases | Use case production services with fluctuating traffic |
| Explicit control over replicas | Autoscaling logic controls replicas |

Kubernetes supports automatic scaling using a [HorizontalPodAutoscaler](/docs/concepts/workloads/autoscaling/horizontal-pod-autoscale/) (HPA).

> If a HPA is enabled for a Deployment, the HPA may override replica counts that you have set manually.

Reviewer 2: ```suggestion
If an HPA is enabled for a Deployment, the HPA may override replica counts that you have set manually.
```

---

> reviewers:

Reviewer 4: ```suggestion
```

title: Scale a Deployment Manually
content_type: Task
weight: 90
---

<!-- overview -->

This page shows how to manually scale a Kubernetes Deployment, and explains when manual scaling is appropriate compared to automatic scaling.

## {{% heading "prerequisites" %}}

{{< include "task-tutorial-prereqs.md" >}} {{< version-check >}}

<!-- lesson content -->

## Create a sample Deployment

If you do not already have a Deployment, follow the instructions in [Run a Stateless Application Using a Deployment](/content/en/docs/tasks/run-application/run-stateless-application-deployment/) to create one.

Alternatively, you can create a simple Deployment for testing.

```shell
kubectl create deployment nginx --image=nginx
```

> To verify that the Deployment was created,run.

Reviewer 4: (nit)
```suggestion
To verify that the Deployment was created, run:
```

### Scale a Deployment to zero

Scale to zero set the replica count of a workload to 0.

```yaml
spec:
  replicas: 2
```

To stop all Pods managed by a Deployment:

```shell
kubectl scale deployment <deployment_name> --replicas=0
```

The output similar to this:

```yaml
apiVersion: apps/v1
kind: Deployment
metadata:
  name: nginx-deployment
spec:
  selector:
    matchLabels:
      app: nginx
  replicas: 0 # Update the replicas from 4 to 0
  template:
    metadata:
      labels:
        app: nginx
    spec:
      containers:
      - name: nginx
        image: nginx:1.16.1
        ports:
        - containerPort: 80
```

Scaling to zero keeps the Deployment object but removes all running Pods.

## Manual scaling versus automatic scaling

| Manual scaling | Automatic scaling |
|----------------|-------------------|
| Developer updates replica count | Kubernetes adjusts replicas based on metrics |
| Predictable or temporary workload changes | Variable or unpredictable workloads |
| Use case load testing, planned traffic increases | Use case production services with fluctuating traffic |
| Explicit control over replicas | Autoscaling logic controls replicas |

> Kubernetes supports automatic scaling using a [HorizontalPodAutoscaler](/docs/concepts/workloads/autoscaling/horizontal-pod-autoscale/)(HPA). If a HPA is enabled for a Deployment, it may override manually set replica counts.

Reviewer 4: (nit)
```suggestion
Kubernetes supports automatic scaling using a [HorizontalPodAutoscaler](/docs/concepts/workloads/autoscaling/horizontal-pod-autoscale/) (HPA).
If a HPA is enabled for a Deployment, the HPA may override replica counts that
you have set manually.
```

kubectl get deployments
```

The output is similar to this:

```
NAME               READY   UP-TO-DATE   AVAILABLE   AGE
nginx              1/1     1            1           10s
```

## Scaling a Deployment manually

Manual scaling adjusts the number of running Pods for a workload by using commands such as `kubectl scale`.

### Scale up a Deployment

Scaling up increases the capacity of an application to handle additional load.

To increase the number of replicas:

```shell
kubectl scale deployment <deployment_name> --replicas=4
```

{{< note >}}
Replace <deployment_name> with the name of your Deployment (for example, nginx).
{{</ note >}}

To verify the change in the Deployment:

```shell
kubectl get deployment <deployment_name>
kubectl get pods
```

The output is similar to this:

```
NAME    READY   UP-TO-DATE   AVAILABLE   AGE
nginx   4/4     4            4           3m16s
```
```
NAME                     READY   STATUS    RESTARTS   AGE
nginx-56c45fd5ff-xxxxx   1/1     Running   0          60s
nginx-56c45fd5ff-xxxxx   1/1     Running   0          60s
nginx-56c45fd5ff-xxxxx   1/1     Running   0          60s
nginx-56c45fd5ff-xxxxx   1/1     Running   0          60s
```

> When the replica count is increased from 1 to 4, Kubernetes creates two additional Pods to match the desired state and schedules them to available nodes while keeping the existing Pods unchanged.

Reviewer: When the replica count is increased from 1 to 4, Kubernetes creates three additional Pods to match the desired state and schedules them to available nodes while keeping the existing Pod unchanged.

```suggestion
When the replica count is increased from 1 to 4, Kubernetes creates three additional Pods to match the desired state and schedules them to available nodes while keeping the existing Pod unchanged.
```

### Scale down a Deployment

Scaling down decreases the capacity of an application by decreasing the number of running Pods, allowing Kubernetes to terminate excess capacity while maintaining the desired state.

To reduce the number of replicas:

```shell
kubectl scale deployment <deployment_name> --replicas=2
```

{{< note >}}
Replace <deployment_name> with the name of your Deployment (for example, nginx).
{{</ note >}}

To verify the change in the Deployment:

```shell
kubectl get deployment <deployment_name>
kubectl get pods
```

The output is similar to this:

```
NAME    READY   UP-TO-DATE   AVAILABLE   AGE
nginx   2/2     2            2           7m16s
```
```
NAME                     READY   STATUS    RESTARTS   AGE
nginx-56c45fd5ff-xxxxx   1/1     Running   0          4m58s
nginx-56c45fd5ff-xxxxx   1/1     Running   0          7m27s
```

When scaling down, Kubernetes terminates Pods following the Pod termination flow.

### Scale a Deployment to zero

> Scale to zero sets the replica count of a workload to 0.

Reviewer: nit:

```suggestion
Scaling to zero sets the replica count of a workload to 0, stopping all Pods managed by that workload.```

nginx-56c45fd5ff-xxxxx   1/1     Running   0          4m58s
nginx-56c45fd5ff-xxxxx   1/1     Running   0          7m27s
```

When scaling down, Kubernetes terminates Pods following the Pod termination lifecycle.

### Scale a Deployment to zero

Scale to zero sets the replica count of a workload to 0.

To stop all Pods managed by a Deployment:

```shell
kubectl scale deployment <deployment_name> --replicas=0
```

{{< note >}}
Replace <deployment_name> with the name of your Deployment (for example, nginx).
{{</ note >}}

To verify that the Deployment is scaled to zero:

```shell
kubectl get deployment <deployment_name>
kubectl get pods
```

The output is similar to this:

```
NAME    READY   UP-TO-DATE   AVAILABLE   AGE
nginx   0/0     0            0           19m
```
```
No resources found in default namespace.
```

This output shows that the Deployment still exists with zero desired replicas, and that no Pods are running.

## Cleanup/kubectl delete deployment nginx

Delete the Deployment by name:

```
kubectl delete deployment <deployment_name>
```

{{< note >}}
Replace <deployment_name> with the name of your Deployment (for example, nginx).
{{</ note >}}

> To verify the Deploument is terminated:

Author: fixed
